static/app/views/alerts/rules/issue/messagingIntegrationModal.tsx

```diff
@@ -32,6 +33,13 @@ function MessagingIntegrationModal({
   project,
   onAddIntegration,
 }: Props) {
+  useEffect(() => {
+    trackAnalytics('onboarding.messaging_integration_modal_rendered', {
```

Comment:
Usually we don't track analytics when something it rendered - it can result in a lot of excess analytic events if done wrong. Generally it's better to record the analytic on an action instead. I'd recommend firing this event when the button is clicked to open this modal.

If you haven't seen it yet, you can add them on the button very easily: https://develop.sentry.dev/development/analytics/#button-click-analytics

Reply:
+1 to the button clicks - just wanted to also thank you for that link! i've been hoping to find something like that! 🎉

Reply:
@malwilley @saponifi3d We need to track how many users are actually shown this setup button so that we can compare this number to how many people actually click the button and install an integration, I'm not sure if there's a better way than tracking when this button is rendered.

Reply:
🤔 maybe trigger the `onboarding.messaging_integration_modal_rendered` onClick of the modal?

That way you could visualize the usage through:
- Page view
- Modal Open
- Setup Integration

and we'd be able to understand where a user might get stuck trying to setup the integration.

Reply:
Ah I think I misread this earlier-- I can definitely change when `onboarding.messaging_integration_modal_rendered` is logged to be onClick! The issue is with tracking `onboarding.setup_messaging_integration_button_rendered`, because I believe this can only be logged when the button is rendered.